### DISCUSSION TITLE
Energy Management for a 2-Motor BEV using Model-Predictive Control (Project 246)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#83] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project 246](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/blob/main/projects/Energy%20Management%20for%20a%202-Motor%20BEV%20using%20Model-Predictive%20Control)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by moanas5** [comment]
Hi, I am Mohamed-Anas a student at MOHAMMED VI POLYTECHNIC UNIVERSITY and interested in working and collaborating on this project

**Reply by SoufianeLaebouss** [reply]
Hi, Mohamed-Anas if you still interested in the project. I also would like to work on it.

**Comment by martinsadab** [comment]
Hello, I am Martins Adaboh. I am currently studying for a Master degree in Control and Optimisation at Imperial College London.  Thank you for the opportunity to work on this project. Looking forward to collaborating with the team.

**Reply by martinsadab** [reply]
Hello Sir, @robertogl. I wonder if it is okay to begin to ask questions here in order to get myself up to speed with the problem.

**Reply by robertogl** [reply]
Hi @martinsadab,

Yes, ask any questions here!

**Reply by martinsadab** [reply]
![image](https://github.com/mathworks/MATLAB-Simulink-Challenge-Project-Hub/assets/151664463/620bda34-f051-46f6-a0c2-31327e51ae45)
Hi @robertogl what is the meaning of the T and R term from the aerodynamic drag equation. The equation is somewhat unconventional. See the image of the equation attached. 
I am trying to form model equations but I can not make sense of this equation. The simulink model does not explain the terms either.

**Reply by jrodgersMW** [reply]
If you scroll down a bit more you can see the definitions for T and R. 

<img width="425" alt="image" src="https://github.com/mathworks/MATLAB-Simulink-Challenge-Project-Hub/assets/64969453/68494acd-fe39-4e42-bb9b-ceca6fed9087">

**Reply by martinsadab** [reply]
Thank you for your prompt response. Got it now!

However, I still do not fully understand the way the block is implemented. 
This is the standard equation for drag force in the literature
![image](https://github.com/mathworks/MATLAB-Simulink-Challenge-Project-Hub/assets/151664463/36f3303c-7857-4c21-9ae0-e873fec344ac)

The simulink block does not capture the square of the vehicle velocity as is common. Could you help reconcile the two equations?

**Reply by jrodgersMW** [reply]
The full set of equations are here:
![image](https://github.com/mathworks/MATLAB-Simulink-Challenge-Project-Hub/assets/64969453/4b0c8b40-5dc2-40ab-aeb9-c25917c92a78)

Basically the additional terms (P/RT) is the ideal gas law and is used to calculate air density. Doing it this way let's us capture the effect of changing temperature, gas, or operating pressure (different planet for example) without knowing the density. The vehicle speed is included if you look at the above equations. We consider the velocity speed in all directions as well as any input relative windspeed. If the vechile is going straight with no extra wind, w=xdot (vehicle speed).  Hope this helps.

**Reply by martinsadab** [reply]
Yes it does. Thank you again.

**Comment by moanas5** [comment]
Hello Mr. Martin, i hope all is good, i believe you will bring to our team a better solution ad you will provide a good work, looking forward for collaboration.

Best,
Mohamed-Anas,
________________________________

**Reply by martinsadab** [reply]
Thank you for the kind words and warm welcome.

**Comment by hgangwar** [comment]
Hello, I'm Harsh Gangwar, a student from Arizona State University. Currently, I'm working on this collaborative project with my team.

**Comment by Thruster55** [comment]
Hi, my name is Harshit Shah and I am a student at Otto Von Guericke University and If anyone is interested in working and collaborating on this project , it would be an interesting insight in control systems .

**Comment by chaotic-fantasyam** [comment]
Where can I find the help documentation for the vehicle Simulink model generated by Virtual Vehicle Composer? I have questions about the generated model import parameters, such as what is Evn?
![Uploading image.png…]()

**Reply by kevinoshiro** [reply]
Hello, 

Virtual Vehicle Composer documentation is located here:
https://www.mathworks.com/help/vdynblks/ref/virtualvehiclecomposer-app.html

Not sure what 'Evn' is.  Env is the prefix for the Environment parameters such as gravity, ambient temperature / pressure.  You can send a screenshot if needed.

Thank you

**Reply by chaotic-fantasyam** [reply]
it's very helpful thank you very much

**Comment by Lensspieler** [comment]
Hello I am a student of RPTU Germany of the course Commercial Vehicle Technology, Please let me know if anyone wants to collaborate for this project.

**Reply by VictorUhr** [reply]
Hello, 
I am interested! I am also studying in RPTU.

**Reply by Thruster55** [reply]
Hey Guys if possible I would also like to collaborate , I am studying in OVGU

**Comment by kasiviswap** [comment]
hi there, i am new to this project i just run the build virtual model, then run simulation model but ,  am getting an error "Unable to create mex function 'BMSBalancingLogic_sfun.mexmaca64' required for simulation.
Component:Simulink | Category:Block diagram error", 


can anyone help to proceed further

**Reply by KodAdi0** [reply]
You sould change your compiler to Mingw64. You can ask LLM's to how to do it. Of course you should download it from add-ons if you dont have it
